test/utils/fin.ts

```diff
@@ -0,0 +1,171 @@
+                manifest = deepmerge(manifest, {
+                    startup_app: sanitizedIdentity
+                });
+            }
+
+            if(!apps.has(identity.uuid)) {
+                const app = new FakeApplication(sanitizedIdentity, manifest);
+                apps.set(identity.uuid, app);
+
+                if(parentUuid) {
+                    app.setParentUuid(parentUuid);
+                }
+
+                emitter.emit('application-created', identity);
+
+                return app;
+            }
+        },
+        wrapSync: (identity: Identity) => {
+            const app = apps.get(identity.uuid);
+
+            if(app){
+                return app;
+            }
+
+            throw new Error(`App ${identity.uuid} / ${identity.name} does not exist.`)
+        }
+    },
+    Window: {
+        wrapSync: (identity: Identity) => {
+            const app = apps.get(identity.uuid);
+
+            if(app) {
+                const window = app.getWindows().get(identity.name);
+
+                if(window){
+                    return window;
+                } else {
+                    app.createChildWindow(identity.name);
+                }
+            } else {
+                throw new Error(`App ${identity.uuid} / ${identity.name} does not exist`);
+            }
+        }
+    }
+};
+
+interface AppState {
+    isRunning: boolean;
+}
+
+class FakeApplication extends EventEmitter {
+    private _identity: Identity;
+    private _windows: Map<String, FakeWindow> = new Map<String, FakeWindow>();
+    private _parentUuid: string|undefined;
+    private _manifest: any;
+    private _state: AppState = {isRunning: true};
+
+    constructor(identity: Identity, manifest: any) {
+        super();
+
+        this._identity = identity;
+        this._manifest = manifest;
+
+        this.createChildWindow(this._identity.name);
+    }
+
+    private updateState(state: AppState) {
+        this._state = Object.assign(this._state, state);
+    }
+    
+    public async getInfo() {
+        return {
+            manifest: this._manifest,
+            parentUuid: this._parentUuid
+        }
+    }
+    
+    public createChildWindow(name: string) {
+        const window = new FakeWindow({uuid: this._identity.uuid, name});
+        this._windows.set(name, window);
+
+        return window;
+    }
+
+    public createChildApp(identity: Identity) {
+        return fin.Application.create(identity, this._manifest, this._identity.uuid);
+    }
+    
+    public getWindows() {
+        return this._windows;
+    }
+    
+    public setParentUuid(parentUuid: string) {
+        this._parentUuid = parentUuid;
+    }
+
+    public close() {
+        this.updateState({isRunning: false});
+        this.emit('closed', {type: "closed", topic: "application", uuid: this._identity.uuid});
+    }
+    
+    public get parentUuid() {
```

Comment:
By convention, we typically put getters/setters between the constructor and member functions.

Reply:
Fixed